### PR TITLE
Add support of missing arguments in `dpnp.count_nonzero`

### PR DESCRIPTION
**Pull request by antonwolfy**
The PR is about to fully get rid of fallback on numpy in `dpnp.count_nonzero` function. It requires to implement support of remaining arguments there.
Another added improvement here is to calculate the result using kernel submitting to a device rather than copying input data to shared USM memory and performing computing on the host.

The implementation is done through `dpnp.sum` call for an input array casted to bool type.
It is enough to demonstrate a huge performance improvements for the `dpnp.count_nonzero` call time. Meanwhile there is possible future improvement by implementation a SYCL extension with separate kernel specific to `count_nonzero` operation.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/dpnp_iface_counting.py [hunk 1]
```diff
@@ -39,48 +39,54 @@
...
+def count_nonzero(a, axis=None, *, keepdims=False):
     """
-    Counts the number of non-zero values in the array ``in_array1``.
+    Counts the number of non-zero values in the array `a`.
 
     For full documentation refer to :obj:`numpy.count_nonzero`.
 
+    Returns
+    -------
+    count : dpnp.ndarray
```
Comment (vtavana) [inline dpnp/dpnp_iface_counting.py:55]: ?
```suggestion
    out : dpnp.ndarray
```

File: dpnp/dpnp_iface_counting.py [hunk 2]
```diff
@@ -39,48 +39,54 @@
...
     --------
     >>> import dpnp as np
-    >>> np.count_nonzero(np.array([1, 0, 3, 0, 5])
-    3
-    >>> np.count_nonzero(np.array([[1, 0, 3, 0, 5],[0, 9, 0, 7, 0]]))
-    5
+    >>> np.count_nonzero(np.eye(4))
+    array(4)
+    >>> a = np.array([[0, 1, 7, 0],
+    ...               [3, 0, 2, 19]])
```
Comment (vtavana) [inline dpnp/dpnp_iface_counting.py:77]:
```suggestion
                   [3, 0, 2, 19]])
```

**Review by vtavana — COMMENTED** [review]
In addition, could we add all data types to `tests/test_counting.py::test_count_nonzero`?

**Comment by antonwolfy** [comment]
> In addition, could we add all data types to `tests/test_counting.py::test_count_nonzero`?

Done

**Review by vtavana — APPROVED** [review]
Thank you @antonwolfy

**Pull request merged**